### ISSUE TITLE
Nav parameter for preventing focus loss by pressing ImGuiKey_Escape?

### ISSUE THREAD

**Issue by Sk44rt** [ocornut/imgui#8059]
### Details:

**My Issue/Question:**

I need to _disable_ the loss of navigation focus when the ImGuiKey_Escape button is pressed
I've been looking for a solution for the last four days. but didn't find anything
then I started looking at what lines in `imgui.cpp` could be commented out so that the focus would not be lost
and i found it!
https://github.com/ocornut/imgui/blob/c4bc6744824de148c3f825ffdcde785510e208ac/imgui.cpp#L13347
if g.NavId is not set to 0 then navigation focus remains

I think we can add bool parameter to ImGuiContext for preventing focus lost
something like `NavPermanentFocus`
and code:
```
if (!g.NavPermanentFocus)
    g.NavId = 0
```
and also, with this solution, focus can move to other items/windows/popups

**Comment by ocornut** [comment]
I thought there was an issue dedicated to this but I cannot find it..
However some related issues: #3200 #2048 #1074
They are related because I think we need a design that covers all those options, even if e.g. #3200 is asking for something that goes the opposite way of your request.

I do agree we need a solution.

@Sk44rt To gather some context, would you mind writing some more about your usage?
Can you confirm that you essentially want keyboard highlight to always be visible, aka essentially ImGui will be the only user of keyboard inputs in your application?

**Comment by Sk44rt** [comment]
> I thought there was an issue dedicated to this but I cannot find it.. However some related issues: #3200 #2048 #1074 They are related because I think we need a design that covers all those options, even if e.g. #3200 is asking for something that goes the opposite way of your request.
> 
> I do agree we need a solution.
> 
> @Sk44rt To gather some context, would you mind writing some more about your usage? Can you confirm that you essentially want keyboard highlight to always be visible, aka essentially ImGui will be the only user of keyboard inputs in your application?

ImGui is the only user of keyboard inputs only when any ImGui window is opened

**Some context**
I'm trying to make a [Kodi](https://kodi.tv/)-like application specifically for usage on big screens (e.g. TV), so Keyboard/Gamepad will be the only way to navigate the UI (for which ImGui is used)
and losing focus is the only problem I've encountered

**Comment by ocornut** [comment]
I have added a few things related to those requests:

- Nav: added `io.ConfigNavCursorVisibleAuto` and `io.ConfigNavCursorVisibleAlways` config options (ab9ce2a).
- Nav: added `SetNavCursorVisible()` to override current state (634a7ed9).

Technically I think the config options are sufficient for most people and would solve this, but `SetNavCursorVisible()` may also occasionally be useful (e.g. during startup to present both modes, while staying in Auto mode).

Other tangentially related options:
- Nav: added `io.ConfigNavEscapeClearFocusItem` (7a56b41).
- Nav: added `io.ConfigNavEscapeClearFocusWindow` (b001038).
- And a couple of related fixes for all the above.

I'm closing this and other related issues as normally solved, but feel free to comment or open new issues if you have question or problem related to those.

(

Also note, if you have mods using internals, that those two fields have been renamed internally:
- Terminology has been changed to call this the "nav cursor" more consistently.
- bool NavDisableHighlight  -> became bool NavCursorVisible (with opposite value). 
- bool NavDisableMouseHover -> became bool NavDisableMouseHover

)
